src/lib/Linearize.hs

```diff
@@ -319,6 +321,14 @@ linearizeDecls (Nest (Let b (DeclBinding ann _ expr)) rest) cont = do
 linearizeExpr :: Emits o => Expr i -> LinM i o Atom Atom
 linearizeExpr expr = case expr of
   Atom x -> linearizeAtom x
+  App (Var f) xs -> do
+    f' <- substM f
+    lookupCustomRules f' >>= \case
+      Nothing -> error "not implemented"
```

Comment:
Is this really "not implemented"?  Could you say what, exactly, is not implemented?

I assume this case mostly doesn't happen because the way an `App (Var f)` would happen here is only if simplify didn't inline the function, and the only way that would happen is if there were rules for it, and right now the only rules are linearization rules; but it would still be good to have a reminder of what this case means.

Reply:
It's the same message that we had before. Linearization runs on simplified code, so most applications should be gone by now. The only exceptions are: (1) the newly added custom linearizations (2) `@noinline` functions. We still don't support differentiation of (2), which is why it's not implemented.